src/Microsoft.AspNetCore.SystemWebAdapters.SessionState/RemoteSession/Framework/GetWriteableSessionHandler.cs

```diff
@@ -0,0 +1,69 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web;
+using System.Web.SessionState;
+using Microsoft.AspNetCore.SystemWebAdapters.SessionState.Serialization;
+
+namespace Microsoft.AspNetCore.SystemWebAdapters.SessionState.RemoteSession;
+
+internal sealed class GetWriteableSessionHandler : HttpTaskAsyncHandler, IRequiresSessionState
+{
+    private const byte EndOfFrame = (byte)'\n';
+
+    private readonly ISessionSerializer _serializer;
+    private readonly ILockedSessionCache _cache;
+
+    public GetWriteableSessionHandler(ISessionSerializer serializer, ILockedSessionCache cache)
+    {
+        _serializer = serializer;
+        _cache = cache;
+    }
+
+    public override async Task ProcessRequestAsync(HttpContext context)
+    {
+        using var timeoutCts = new CancellationTokenSource(TimeSpan.FromMinutes(context.Session.Timeout));
+        using var cts = CancellationTokenSource.CreateLinkedTokenSource(timeoutCts.Token, context.Response.ClientDisconnectedToken);
+
+        await ProcessRequestAsync(new HttpContextWrapper(context), cts.Token).ConfigureAwait(false);
```

Comment:
Why wrap the context with HttpContextWrapper?

Reply:
Because that is mockable. HttpContext itself cannot be mocked in any way

Reply:
Ah, interesting. Good to know.